homeassistant/components/myneomitis/select.py

```diff
@@ -0,0 +1,260 @@
+        hass (HomeAssistant): The Home Assistant instance.
+        config_entry (MyNeomitisConfigEntry): The configuration entry.
+        async_add_entities (AddEntitiesCallback): Callback to add entities.
+
+    """
+    api = config_entry.runtime_data.api
+    devices = config_entry.runtime_data.devices
+
+    def _get_entity_description(device: dict) -> MyNeoSelectEntityDescription | None:
+        """Determine the appropriate entity description for a device."""
+        if device["model"] == "EWS":
+            # Check if device has relayMode to determine if it's a relais or pilote
+            if "relayMode" in device.get("state", {}):
+                return SELECT_TYPES["relais"]
+            return SELECT_TYPES["pilote"]
+        if device["model"] == "UFH":
+            return SELECT_TYPES["UFH"]
+        return None
+
+    def _create_entity(device: dict) -> MyNeoSelect | None:
+        """Create a select entity for a device.
+
+        Args:
+            device: The device data dictionary.
+
+        Returns:
+            A MyNeoSelect entity or None if the device is not supported.
+
+        """
+        description = _get_entity_description(device)
+        if description is None:
+            return None
+        return MyNeoSelect(api, device, devices, description)
+
+    select_entities = [
+        entity
+        for device in devices
+        if device["model"] in MODELES or device["model"] in SUB_MODELES
+        if (entity := _create_entity(device)) is not None
+    ]
+
+    async_add_entities(select_entities)
+
+
+class MyNeoSelect(SelectEntity):
+    """Select entity for MyNeomitis devices."""
+
+    entity_description: MyNeoSelectEntityDescription
+
+    def __init__(
+        self,
+        api: PyAxencoAPI,
+        device: dict[str, Any],
+        devices: list[dict[str, Any]],
+        description: MyNeoSelectEntityDescription,
+    ) -> None:
+        """Initialize the MyNeoSelect entity.
+
+        Args:
+            api: The API instance to communicate with the device.
+            device: The device data dictionary.
+            devices: A list of all devices managed by the integration.
+            description: The entity description for this select entity.
+
+        """
+        super().__init__()
+        self.entity_description = description
+        self._api = api
+        self._device = device
+        self._attr_name = f"MyNeo {device['name']}"
+        self._attr_unique_id = f"myneo_{device['_id']}"
+        self._attr_available = device["connected"]
+        self._is_sub_device = device["model"] in SUB_MODELES
+
+        # Set current option based on device state
+        current_mode = device.get("state", {}).get(description.state_key)
+        self._attr_current_option = description.reverse_preset_mode_map.get(
+            current_mode, STATE_UNKNOWN
+        )
+
+        self._program = device.get("program", {}).get("data", {})
+
+        api.register_listener(device["_id"], self.handle_ws_update)
```

Comment:
2 things
1. We should register this in `async_added_to_hass`. Because if an entity is disabled, we should not subscribe or write state.
2. We should also unregister listeners in `async_will_remove_from_hass`. This way if someone disables an entry (or reloads the integration), everything is cleaned up. You can also pass a cleanup method to `self.async_on_remove()`